ufl/algorithms/formtransformations.py

```diff
@@ -446,7 +447,7 @@ def compute_energy_norm(form, coefficient):
         if coefficient.ufl_function_space() != U:
             error("Trying to compute action of form on a "
                   "coefficient in an incompatible element space.")
-    return replace(form, {u: coefficient, v: coefficient})
+    return Action(Action(form, coefficient), coefficient)
```

Comment:
Should be lower-case `action`, rather than constructing the object. Necessary because `replace` doesn't work on a `Matrix` where we want to produce `<c| M c>`